### DISCUSSION TITLE
Порядковый номер

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#103] [Как это сделать?] [answered]
Как сделать чтоб порядковый номер в строке обнулялся при наступлении нового года? Слева "Год", справа "Порядковый номер".
![image](https://user-images.githubusercontent.com/103171127/162564402-369b15e3-6689-4732-afe1-24a262ce294b.png)
Вот тут 22 год. Когда станет год 23, чтоб счёт начинался заново.
![image](https://user-images.githubusercontent.com/103171127/162564429-924018d1-eb8a-4ce6-bc58-559dc0802f47.png)
Я считал по id. Как правильно реализовать?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Вот разбиралось присвоение порядкового номера: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/51

**Но в этом посте пример с НЕИЗМЕНЯЕМЫМИ поле добавления номерами.**

Также будет полезно это: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/88

Суть решения — технически разделить год и порядковый номер в разные поля:

**Год:**

```
=: dateFormat(date: $#nd; format: "y")

// --!->> Execute only on adding <<-!--
```

**Номер:**

```

=: select(table: $#ntn; field: 'num'; where: 'year' = #year;  order: 'id' desc) + 1


--!->> Execute only on adding <<-!--
```

**Итоговый номер:**

```
=: str`#year + "-" + #num`

// --!->> Execute only on adding <<-!--
```

<img width="336" alt="Снимок экрана 2022-04-09 в 18 09 29" src="https://user-images.githubusercontent.com/55755565/162580016-49cf9aed-66f5-4906-b6a1-fea2e35ba6b5.png">
